### PR TITLE
Updated key signature

### LINKED ISSUE #15: Order of sharps or flats in key signature
We should decide if we want ordered sharps/flats in the key signature or allow arbitrary sequences. For example: do we tolerate D major written as $xCF or it must be $xFC?

I found 225 cases with wrong order of sharps/flats (including cases like $Bb), see [attached list](https://github.com/rism-digital/pae-code-spec/files/7337982/keysig_wrong_notes_order.csv). The real number of those could be slightly higher than this, since grammatically wrong key signatures are excluded from the analysis.

The actual paec-grammar version does not check the order of sharps/flats.

### LINKED ISSUE #16: Repeated sharps or flats in key signature
I found 76 cases with repeated sharps/flats in the key signature (see [list](https://github.com/rism-digital/pae-code-spec/files/7339684/keysig_multiple_notes.csv), the real case number might be higher). Examples are: $bBEE, $xFCGF, but also $xFCG[G]

Should we correct those entries and modify the paec-grammar in order to detect them?

What about $xFCG[G] ? We probably have no chance to know which G is too much

### LINKED ISSUE #17: Clarify use of square brackets with more accidentals in key signature
The spec should explicitly state that key signatures with consecutive accidentals requiring square brackets must be included within a single square bracket. Consecutive square brackets are not allowed. For example: $xF[CG] is allowed, $x[F]C[G] is allowed, but $xF[C][G] is forbidden.

### LINKED ISSUE #26: Clarify whether `n` is a valid accidental type in key signatures.
Not sure if this is a Muscat thing, a Verovio thing, or a P&E thing, so I'll start discussion here.
The guidelines say: 
```
12. Change of clef, key or time signature:
Use % to change the clef, $ to change the key, and @ to change the time signature. Follow this with the new indication (clef, key, or time), followed by a space.

Examples:
%C-1 '2A
%C-1 $xFC '8B
@3/2 '1C
$nBE $xFC
```

I'm interested in the last example. With the new Verovio validation, if you enter something like `4CDEF/$nBE $xFC AB''C` then it's not happy because you can't change the key signature more than once in a measure: 
image

But adding the required bar line `4CDEF/$nBE /$xFC AB''C` isn't very satisfying optically:
image

But maybe you can just leave out the cancelled key signature and go directly to the new one:
`4CDEF/$xFC AB''C`
image

The previous Verovio version (still visible on muscat-training) didn't print the cancelled key signature anyways, so I guess we haven't missed this feature. I am wondering if Verovio should be adjusted, or if we can make a shortcut in the guidelines to go directly to the new key signature (analogue to going directly to the new clef or meter).

### LINKED ISSUE #36: Make key signature a recommended field
Currently the key signature can be empty. However, this makes it unclear whether the key signature is missing, or whether it is a completely natural key signature.

The section should be clarified to say that encoders **should** provide a value, and that a value of `n` can be given if no key signature is present.

It should also clarify that parsers should assume a value of `n` if no key signature is present.

Split off from #11 
Related to #26

### PR DESCRIPTION
**Pull request by ahankinson**
Replaces #54 

Fixes #15
Fixes #16
Fixes #17
Fixes #26
Refs #35
Fixes #36

### CODE REVIEW

File: v2/index.html [hunk 1]
```diff
@@ -180,21 +188,42 @@ <h3>Clef</h3>
...
-        <div class="issue" data-number="26"></div>
         <p>
-            The symbol <code>x</code> indicates sharp keys and <code>b</code> flat keys. The symbol is followed by the
-            capital letters that indicate the altered notes.
+            The character <code>x</code> indicates sharp keys and <code>b</code> flat keys. These characters MUST be
+            followed by a list of <a href="#note-names">note names</a> that indicate the altered notes.
         </p>
-
+        <p>
+            The list of note names in a key signature SHOULD follow the <a>circle of fifths</a> ordering. Missing
```
Comment (lpugin) [inline v2/index.html:204]:
```suggestion
            The list of note codes in a key signature SHOULD follow the <a>circle of fifths</a> ordering. Missing
```

File: v2/index.html [hunk 2]
```diff
@@ -180,21 +188,42 @@ <h3>Clef</h3>
...
+            accidentals in this list SHOULD be supplied by the <a>transcriber</a>. Note names MUST NOT be repeated.
```
Comment (lpugin) [inline v2/index.html:205]:
```suggestion
            accidentals in this list SHOULD be supplied by the <a>transcriber</a>. Note codes MUST NOT be repeated.
```

File: v2/index.html [hunk 3]
```diff
@@ -180,21 +188,42 @@ <h3>Clef</h3>
...
+        </p>
+        <p>
+            A key signature containing a single <code>n</code> character MAY be supplied to indicate a natural key
+            signature. This character MUST NOT be followed by any note names.
```
Comment (lpugin) [inline v2/index.html:209]:
```suggestion
            signature. This character MUST NOT be followed by any note codes.
```

File: v2/index.html [hunk 4]
```diff
@@ -180,21 +188,42 @@ <h3>Clef</h3>
...
+        </p>
+        <p>
+            A key signature MAY contain note names within square brackets, <code>[]</code>, to indicate that the
```
Comment (lpugin) [inline v2/index.html:212]:
```suggestion
            A key signature MAY contain note codes within square brackets, <code>[]</code>, to indicate that the
```

File: v2/index.html [hunk 5]
```diff
@@ -180,21 +188,42 @@ <h3>Clef</h3>
...
+            note names are not in the original source and have been supplied by the <a>transcriber</a>. Consecutively
```
Comment (lpugin) [inline v2/index.html:213]:
```suggestion
            note codes are not in the original source and have been supplied by the <a>transcriber</a>. Consecutively
```

File: v2/index.html [hunk 6]
```diff
@@ -180,21 +188,42 @@ <h3>Clef</h3>
...
+            supplied note names MUST be within a single set of brackets. A key signature MAY contain more than one
```
Comment (lpugin) [inline v2/index.html:214]:
```suggestion
            supplied note codes MUST be within a single set of brackets. A key signature MAY contain more than one
```

**Review by lpugin — COMMENTED** [review]
I am suggesting to use `note codes` instead of `note names`. But only as a suggestion.

**Comment by ahankinson** [comment]
Since we would also need to change it in the actual "Note Names" section, outside of the scope of this change, I have opened #74 for this.

**Review by lpugin — APPROVED** [review]

**Pull request merged**